docs/recipes/LocalFirstDataWithPowerSync.md

```diff
@@ -0,0 +1,713 @@
+```
+
+:::tip
+For more information on authentication with Supabase, refer to the [Supabase Auth documentation](https://supabase.io/docs/guides/auth).
+:::
+
+### Initializing the PowerSync Instance, and providing it to the app
+
+We need a sinple point of connection to the PowerSync instance that we can interact with throughout the app. 
+
+To achieve this we create a `Database` class that encompasses both the PowerSync and Supabase configurations. 
+
+Then we provide a stable reference to the `Database` instance through a React Context.
+
+```tsx
+// app/services/database.ts
+import '@azure/core-asynciterator-polyfill';
+import 'react-native-polyfill-globals/auto';
+import React, {PropsWithChildren} from "react";
+import {AbstractPowerSyncDatabase, RNQSPowerSyncDatabaseOpenFactory} from '@journeyapps/powersync-sdk-react-native';
+import {SupabaseConnector, supabaseConnector} from "./supabase"; // Adjust the path as needed
+import {AppSchema} from './schema'; // Adjust the path as needed
+
+export class Database {
+    // We expose the PowerSync instance and the Supabase connector for easy access elsewhere in the app
+    powersync: AbstractPowerSyncDatabase;
+    supabase: SupabaseConnector;
+
+    /**
+     * Initialize the Database class with a new PowerSync instance
+     */
+    constructor() {
+        const factory = new RNQSPowerSyncDatabaseOpenFactory({
+            schema: AppSchema,
+            dbFilename: 'sqlite.db',
+        });
+        this.powersync = factory.getInstance();
+        // include the Supabase connector for easy access
+        this.supabase = supabaseConnector;
+    }
+
+    /**
+     * Initialize the PowerSync instance and connect it to the Supabase backend.
+     * This will call `fetchCredentials` on the Supabase connector to get the session token.
+     * So if your database requires authentication, the user will need to be signed in before this is called.
+     */
+    async init() {
+        await this.powersync.init();
+        // Connect the PowerSync instance to the Supabase backend through the connector
+        await this.powersync.connect(supabaseConnector);
+    }
+}
+
+// Then we create an instance of the Database class and create a context for it to hold a stable reference
+export const database = new Database();
+
+export const DatabaseContext = React.createContext<Database | undefined>(undefined);
+
+export const useDatabase = () => {
+    const context: Database | undefined = React.useContext(DatabaseContext);
+    if (!context) {
+        throw new Error("useDatabase must be used within a DatabaseProvider");
+    }
+    return context;
+};
+
+// Finally, we create a provider component that initializes the database and provides it to the app
+export function DatabaseProvider({children}: PropsWithChildren<{}>) {
+    useEffect(() => {
+        database.init().catch(console.error);
+    }, []);
+
+    return (
+        <DatabaseContext.Provider value={database}>
```

Comment:
The file should be `.tsx` that's probably the problem.

Also use this implementation:

```tsx
import React, {PropsWithChildren} from "react";
import {AbstractPowerSyncDatabase, RNQSPowerSyncDatabaseOpenFactory} from '@journeyapps/powersync-sdk-react-native';
import {SupabaseConnector, supabaseConnector} from "./supabase"; // Adjust the path as needed
import {AppSchema} from './schema'; // Adjust the path as needed

export class Database {
    // We expose the PowerSync instance and the Supabase connector for easy access elsewhere in the app
    powersync: AbstractPowerSyncDatabase;
    supabase: SupabaseConnector;

    /**
     * Initialize the Database class with a new PowerSync instance
     */
    constructor() {
        const factory = new RNQSPowerSyncDatabaseOpenFactory({
            schema: AppSchema,
            dbFilename: 'sqlite.db',
        });
        this.powersync = factory.getInstance();
        // include the Supabase connector for easy access
        this.supabase = supabaseConnector;
    }

    /**
     * Initialize the PowerSync instance and connect it to the Supabase backend.
     * This will call `fetchCredentials` on the Supabase connector to get the session token.
     * So if your database requires authentication, the user will need to be signed in before this is called.
     */
    async init() {
        await this.powersync.init();
        // Connect the PowerSync instance to the Supabase backend through the connector
        await this.powersync.connect(supabaseConnector);
    }
}

// Then we create an instance of the Database class and create a context for it to hold a stable reference
export const database = new Database();

export const DatabaseContext = React.createContext<Database | undefined>(undefined);

export const useDatabase = () => {
    const context: Database | undefined = React.useContext(DatabaseContext);
    if (!context) {
        throw new Error("useDatabase must be used within a DatabaseProvider");
    }
    return context;
};

// Finally, we create a provider component that initializes the database and provides it to the app
export function DatabaseProvider({children}: PropsWithChildren<{}>) {
    useEffect(() => {
        database.init().catch(console.error);
    }, []);

    return (
        <DatabaseContext.Provider value={database}>
            <PowerSyncContext.Provider value={database.powersync}>
                {children}
            </PowerSyncContext.Provider>
        </DatabaseContext.Provider>
    );
}
```